01-data-structures/04-hashes-part-1/hashclass.rb

```diff
@@ -1,27 +1,59 @@
 class HashClass
+  require_relative 'hash_item'
 
   def initialize(size)
     @items = Array.new(size)
+    @size = size
   end
 
   def []=(key, value)
+    #if nil adds new hash item
+    if @items[index(key, @items.length)].nil?
+      @items[index(key, @items.length)] = HashItem.new(key, value)
+    #if the keys don't match, and out of room, resize 
+    elsif @items[index(key, @items.length)].key != key
+      self.resize
+      self[key] = value
+    #if there is a collision expand the array
+    elsif @items[index(key, @items.length)].key === key && @items[index(key, @items.length)].value != value
+      self.resize
+      @items[index(key, @items.length)].value = value
+    end
   end
 
 
   def [](key)
+    item = @items[index(key, @items.length)]
+    #if item is nil return nil, else return value of item
+    item.nil? ? nil : item.value
   end
 
   def resize
+    #takes size of hash and doubles it
+    @size = @size * 2
+    #creates new array, at a larger size and assigns it to variable
+    doubled_hash = Array.new(@size)
+    #goes through items and reassigns
+    @items.each do |item|
+      if item != nil
+        doubled_hash[index(item.key, @size)] = item
```

Comment:
You do not have to fix this since it is not tested for, but what if there is already something in `index(item.key, @size)`?